airbyte-integrations/connectors/source-s3/source_s3/v4/legacy_config_transformer.py

```diff
@@ -46,10 +47,11 @@ def convert(cls, legacy_config: SourceS3Spec) -> Mapping[str, Any]:
         return transformed_config
 
     @classmethod
-    def _create_globs(cls, path_pattern: str, path_prefix: str) -> List[str]:
-        if path_prefix:
-            return [path_prefix + path_pattern]
-        return [path_pattern]
+    def _create_globs(cls, path_pattern: str) -> List[str]:
+        if "|" in path_pattern:
```

Comment:
let's add one test case that we separate legacy glob input with the the or separators and confirm the resulting glob is a list